### ISSUE TITLE
Explicit declaring protocol conformance does not elicit error on incorrect member type

### ISSUE THREAD

**Issue by SandyChapman** [microsoft/pyright#3335]
When implementing a protocol explicitly, Pyright fails to check the type of the member variables.

Example:

```python
from dataclasses import dataclass
from typing import Protocol

class MyProtocol(Protocol):
    a: str

@dataclass
class ExplicitlyConformsTo(MyProtocol):
    a: int  # Fails to identify type error here.

@dataclass
class ImplicitlyConformsToMyProtocol:
    a: int

def f(p: MyProtocol):
    ...

f(ExplicitlyConformsTo(a=1))
f(ImplicitlyConformsToMyProtocol(a=1)) # Correctly emits error.
```

Pyright correctly flags the incorrect usage of `ImplicitlyConformsToMyProtocol`, but doesn't emit an error on the declaration of `ExplicitlyConformsToProtocol` or its usage.

I'm not sure if this is intentional, but I couldn't find anything in [PEP-544](https://peps.python.org/pep-0544/) to state this is the expected behaviour.

**Comment by erictraut** [comment]
You need to enable the `reportIncompatibleVariableOverride` check, which is disabled by default.

**Comment by SandyChapman** [comment]
Thanks @erictraut !
